server/consumer.go

```diff
@@ -1520,6 +1531,44 @@ func (o *consumer) processNak(sseq, dseq uint64) {
 			return
 		}
 	}
+	// Check to see if we have delays attached.
+	if len(nak) > len(AckNak) {
```

Comment:
in my mind I think we need this for normal retries too, imagine a poison pill message. It crashes the consumers so they wont be NaK'ing it, the bakoff would minimise the impact of that poison pill significantly

Reply:
If you had a backoff designation it would be in affect for that scenario. That and NAK's being able to delay as well, even across consumer leader changes and server failures I think is a good start.

Reply:
Ah, I missed that this backoff is also triggering for !NaK retries

Reply:
NAK would try redelivery immediately by design, so this allows to express a delay before redelivery for a NAK's message.